packages/opentelemetry-instrumentation-langchain/opentelemetry/instrumentation/langchain/__init__.py

```diff
@@ -36,14 +41,109 @@ def _instrument(self, **kwargs):
         tracer_provider = kwargs.get("tracer_provider")
         tracer = get_tracer(__name__, __version__, tracer_provider)
 
+        traceloopCallbackHandler = TraceloopCallbackHandler(tracer)
         wrap_function_wrapper(
             module="langchain_core.callbacks",
             name="BaseCallbackManager.__init__",
-            wrapper=_BaseCallbackManagerInitWrapper(TraceloopCallbackHandler(tracer)),
+            wrapper=_BaseCallbackManagerInitWrapper(traceloopCallbackHandler),
+        )
+
+        if not self.disable_trace_context_propagation:
+            self._wrap_openai_functions_for_tracing(traceloopCallbackHandler)
+
+    def _wrap_openai_functions_for_tracing(self, traceloopCallbackHandler):
+        openai_tracing_wrapper = _OpenAITracingWrapper(traceloopCallbackHandler)
+
+        # Wrap langchain_community.llms.openai.BaseOpenAI
+        wrap_function_wrapper(
+            module="langchain_community.llms.openai",
```

Comment:
Why not wrap the base class `BaseLLM`? https://github.com/langchain-ai/langchain/blob/316541536940a0e063ed09f48d2bec05ded862c3/libs/core/langchain_core/language_models/llms.py#L292
Then theoretically, it should work for all models

Reply:
I agree that wrapping the base class would be preferable. However, I couldn't find a way to access the `run_id`. For example, when I set a breakpoint in the `invoke()` method while debugging `test_trace_propagation[OpenAI]`, I found that `config["run_id"]` is [empty](https://github.com/langchain-ai/langchain/blob/316541536940a0e063ed09f48d2bec05ded862c3/libs/core/langchain_core/language_models/llms.py#L397). The `run_id` is determined further down the stack trace.

Reply:
Oh well
It's a good start though